sensor/tests/resource/networkpolicy/networkpolicy_test.go

```diff
@@ -99,7 +99,11 @@ func (s *NetworkPolicySuite) Test_Deployment_NetpolViolations() {
 			s.testContext.RunTest(t,
 				helper.WithResources(resourcesToApply),
 				helper.WithTestCase(func(t *testing.T, tc *helper.TestContext, objects map[string]k8s.Object) {
-					tc.LastViolationState(t, "nginx-deployment", checkViolations(testCase.violationsExpected), name)
+					if len(testCase.violationsExpected) == 0 {
+						tc.NoViolationForDeployment(t, "nginx-deployment", "no violations for deployment")
+					} else {
+						tc.LastViolationState(t, "nginx-deployment", checkViolations(testCase.violationsExpected), name)
+					}
```

Comment:
It looks like the existing violations need a moment to disappear.
Shall we maybe retry that block for a while until the conditions are met?

Reply:
Changed the check a little to fix this. Let's see in the next run.